src/Microsoft.ML.Data/Prediction/CalibratorCatalog.cs

```diff
@@ -0,0 +1,438 @@
+        private TICalibrator _calibrator;
+        private readonly string _loaderSignature;
+
+        internal CalibratorTransformer(IHostEnvironment env, TICalibrator calibrator, string loaderSignature)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(CalibratorTransformer<TICalibrator>)))
+        {
+            Host.CheckRef(calibrator, nameof(calibrator));
+
+            _loaderSignature = loaderSignature;
+            _calibrator = calibrator;
+        }
+
+        // Factory method for SignatureLoadModel.
+        internal CalibratorTransformer(IHostEnvironment env, ModelLoadContext ctx, string loaderSignature)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(CalibratorTransformer<TICalibrator>)))
+        {
+            Contracts.AssertValue(ctx);
+
+            _loaderSignature = loaderSignature;
+            ctx.CheckAtModel(GetVersionInfo());
+
+            // *** Binary format ***
+            // model: _calibrator
+            ctx.LoadModel<TICalibrator, SignatureLoadModel>(env, out _calibrator, @"Calibrator");
+        }
+
+        string ISingleFeaturePredictionTransformer<TICalibrator>.FeatureColumn => DefaultColumnNames.Score;
+
+        ColumnType ISingleFeaturePredictionTransformer<TICalibrator>.FeatureColumnType => NumberType.Float;
+
+        TICalibrator IPredictionTransformer<TICalibrator>.Model => _calibrator;
+
+        bool ITransformer.IsRowToRowMapper => true;
+
+        public override void Save(ModelSaveContext ctx)
+        {
+            Contracts.AssertValue(ctx);
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // model: _calibrator
+            ctx.SaveModel(_calibrator, @"Calibrator");
+        }
+
+        private protected override IRowMapper MakeRowMapper(Schema schema) => new Mapper<TICalibrator>(this, _calibrator, schema);
+
+        protected VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "CALTRANS",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: _loaderSignature,
+                loaderAssemblyName: typeof(CalibratorTransformer<>).Assembly.FullName);
+        }
+
+    private sealed class Mapper<TCalibrator> : MapperBase
+        where TCalibrator : class, ICalibrator
+        {
+            private TCalibrator _calibrator;
+            private int _scoreColIndex;
+            private CalibratorTransformer<TCalibrator> _parent;
+
+            internal Mapper(CalibratorTransformer<TCalibrator> parent, TCalibrator calibrator, Schema inputSchema) :
+                base(parent.Host, inputSchema)
+            {
+                _calibrator = calibrator;
+                _parent = parent;
+                inputSchema.TryGetColumnIndex(DefaultColumnNames.Score, out _scoreColIndex);
```

Comment:
>TryGetColumnIndex [](start = 28, length = 17)

You might want to replace TryGetColumnIndex with GetColumnOrNull.

Reply:
This works better here, because of GetDependenciesCore below needing the index. 

Is this on the path to deprecation?

---
In reply to: [240340891](https://github.com/dotnet/machinelearning/pull/1816#discussion_r240340891) [](ancestors = 240340891)

Reply:
It will be deprecated.

---
In reply to: [240426878](https://github.com/dotnet/machinelearning/pull/1816#discussion_r240426878) [](ancestors = 240426878,240340891)